File: Leetcode/643. Maximum Average Subarray I/solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        start = 0
        _sum = sum(nums[start:k])
        max_avg = _sum / k
        n = len(nums)

        for i in range(k, n):
            _sum += nums[i]
            _sum -= nums[start]
            start += 1
            avg = _sum/k
            if avg > max_avg:
                max_avg = avg

        return max_avg

File: Leetcode/643. Maximum Average Subarray I/solution.rs
impl Solution {
    pub fn find_max_average(nums: Vec<i32>, k: i32) -> f64 {
        let k = k as usize;
        let mut count: i32 = nums.iter().take(k).sum();
        let mut max_count = count;

        for i in k..nums.len() {
            count += nums[i] - nums[i - k];
            if count > max_count {
                max_count = count;
            }
        }

        max_count as f64 / k as f64
    }
}